### PR TITLE
[ALLUXIO-2336] Fixed specifier character handling in LsCommand

### PR DESCRIPTION
**Pull request by HorizonNet**
https://alluxio.atlassian.net/browse/ALLUXIO-2336

- Specifier characters used by String.format cause an error in the ls
  shell command
- Fixed this and the ls shell command can now handle specifier
  characters in a file name
- Added test for this behavior

### CODE REVIEW

**Comment by aaudiber** [comment]
Thanks for looking into this @HorizonNet! I think the problem is that filenames containing `%` fail because we use them as format strings in the `ls` method when we do
```java
System.out.format(formatLsString(...
```
This line double-formats the input for no reason (format is called with no arguments except the format string). Instead of `System.out.format` it should be 
```java
System.out.print(formatLsString(...
```

**Comment by HorizonNet** [comment]
@aaudiber Thanks for the comment. I would suggest something in between. Instead of using `format` we should use `print`, but we should stay with the URL encoding, because without it `localhost,61764,1476207067267..meta` would be `localhost%2C61764%2C1476207067267..meta`, which I think is not the intended behavior.

**Comment by aaudiber** [comment]
@HorizonNet I think we want to actually show the `%20` since it's part of the file name. @apc999 is this what you meant in the ticket?

**Comment by HorizonNet** [comment]
@aaudiber Ok, sounds good to me too. Decoding it causes some additional problems in the web UI. Just let me know the direction of the ticket.

**Comment by apc999** [comment]
@HorizonNet thanks for the PR. I was thinking to have "%2" as part of the file name in both webUI (already achieved) and `alluxio fs ls` (exception thrown before your PR). That is my intention in the ticket

**Comment by HorizonNet** [comment]
@apc999 Thanks for the clarification. I updated the PR accordingly.

File: tests/src/test/java/alluxio/shell/command/LsCommandTest.java [hunk 1]
```diff
@@ -249,4 +249,22 @@ public void lsr() throws Exception {
...
+  @LocalAlluxioClusterResource.Config(
+      confParams = {PropertyKey.Name.SECURITY_AUTHORIZATION_PERMISSION_ENABLED, "false",
+          PropertyKey.Name.SECURITY_AUTHENTICATION_TYPE, "NOSASL"})
+  public void lsWithFormatSpecifierCharacter() throws IOException, AlluxioException {
+    String fileName = "/localhost%2C61764%2C1476207067267..meta.1476207073442.meta";
+    FileSystemTestUtils.createByteFile(mFileSystem, fileName, WriteType.MUST_CACHE, 10);
+    URIStatus file = mFileSystem.getStatus(new AlluxioURI(fileName));
+    mFsShell.run("ls", "/");
+    String expected =
+        getLsNoAclResultStr("/localhost%2C61764%2C1476207067267..meta.1476207073442.meta",
```
Comment (aaudiber) [inline tests/src/test/java/alluxio/shell/command/LsCommandTest.java]: lets replace the string with `fileName` to make it easier to tell that they are the same

**Comment by HorizonNet** [comment]
@aaudiber Thanks for the comment. I updated the PR.

**Pull request merged**